pkg/apis/kops/validation/validation.go

```diff
@@ -1113,3 +1117,17 @@ func validateNodeLocalDNS(spec *kops.ClusterSpec, fldpath *field.Path) field.Err
 
 	return allErrs
 }
+
+func validateClusterAutoscaler(cluster *kops.Cluster, spec *kops.ClusterAutoscalerConfig, fldPath *field.Path) (allErrs field.ErrorList) {
+	if !cluster.IsKubernetesGTE("1.15") || cluster.IsKubernetesGTE("1.20") {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "Cluster autoscaler requires kubernetesVersion between 1.11 and 1.19"))
+	}
+
+	allErrs = append(allErrs, IsValidValue(fldPath.Child("expander"), spec.Expander, []string{"least-waste", "random", "most-pods"})...)
+
+	if kops.CloudProviderID(cluster.Spec.CloudProvider) == kops.CloudProviderOpenstack {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "Cluster autoscaler is not supported on OpenStack"))
```

Comment:
I suspect it might not be supported on Spotinst either.

Reply:
I suspect the same. They still roll their own, it seems

Reply:
Doesn't seem like we treat spotinst fully as a cloudprovider. And cluster autoscaler will work for those IGs not managed by Spotinst. So I think it is best to leave it as it is for now. Spotinst is also behind a feature flag, so users know to take care.

Reply:
Fair enough